tests/test_extents.cpp

```diff
@@ -79,25 +142,24 @@ using extents_test_types =
 TYPED_TEST_SUITE(TestExtents, extents_test_types);
 
 TYPED_TEST(TestExtents, rank) {
-  EXPECT_EQ(this->exts.rank(), this->static_sizes.size());
-}
-
-TYPED_TEST(TestExtents, dynamic_rank) {
-  EXPECT_EQ(this->exts.rank_dynamic(), this->dyn_sizes.size());
+  dispatch_host = true;
+  this->test_rank();
+  dispatch_host = false;
+  this->test_rank();
```

Comment:
Meaning we run twice w/o CUDA.
If we always have that pattern, can't we encapsulate it?